packages/core/src/mesh/ModelMesh.ts

```diff
@@ -14,4 +58,783 @@ export class ModelMesh extends Mesh {
     super(engine);
     this.name = name;
   }
+
+  /**
+   * Upload Mesh Data to the graphics API.
+   * @param noLongerAccessible - Whether to access data later. If true, you'll never access data anymore (free memory cache)
+   */
+  uploadMeshData(noLongerAccessible: boolean): void {
+    const { _indicesArray, _vertexSlotChanged, _accessible } = this;
+    if (!_accessible) {
+      throw "Not allowed to access data while accessible is false.";
+    }
+
+    // Structure of vertex buffer has changed
+    if (_vertexSlotChanged || this._vertexCountChanged) {
+      const vertexElements = this._updateVertexElements();
+      const elementCount = this._elementCount;
+      const vertices = new Float32Array(elementCount * this.vertexCount);
```

Comment:
Here you can add a condition of "if(this._verticesArray.length!==elementCount * this.vertexCount)" .
Because the developer maybe remove color and add tangent, In fact the vertices data length is same,do not need to re-construct.